src/Upgrader/Business/Upgrader/Upgrader.php

```diff
@@ -7,63 +7,75 @@
 
 namespace Upgrader\Business\Upgrader;
 
-use Upgrader\Business\Upgrader\Bridge\PackageManagementSystemBridge;
-use Upgrader\Business\Upgrader\Bridge\ReleaseGroupTransferBridgeInterface;
+use Upgrader\Business\Upgrader\Request\UpgraderRequest;
 use Upgrader\Business\Upgrader\Response\Collection\UpgraderResponseCollection;
+use Upgrader\Business\Upgrader\Strategy\ComposerUpdateStrategy;
+use Upgrader\Business\Upgrader\Strategy\ReleaseGroupStrategy;
 use Upgrader\Business\VersionControlSystem\VcsInterface;
 
 class Upgrader implements UpgraderInterface
 {
     /**
-     * @var \Upgrader\Business\Upgrader\Bridge\ReleaseGroupTransferBridgeInterface
+     * @var \Upgrader\Business\Upgrader\Strategy\ComposerUpdateStrategy
      */
-    protected $releaseGroupManager;
+    protected $composerUpdateStrategy;
 
     /**
-     * @var \Upgrader\Business\Upgrader\Bridge\PackageManagementSystemBridge
+     * @var \Upgrader\Business\Upgrader\Strategy\ReleaseGroupStrategy
      */
-    protected $dataProviderManager;
+    protected $releaseGroupStrategy;
 
     /**
      * @var \Upgrader\Business\VersionControlSystem\VcsInterface
      */
     protected $vcs;
 
     /**
-     * @param \Upgrader\Business\Upgrader\Bridge\ReleaseGroupTransferBridgeInterface $releaseGroupManager
-     * @param \Upgrader\Business\Upgrader\Bridge\PackageManagementSystemBridge $dataProviderManager
+     * @param \Upgrader\Business\Upgrader\Strategy\ComposerUpdateStrategy $composerUpdateStrategy
+     * @param \Upgrader\Business\Upgrader\Strategy\ReleaseGroupStrategy $releaseGroupStrategy
      * @param \Upgrader\Business\VersionControlSystem\VcsInterface $vcs
      */
     public function __construct(
-        ReleaseGroupTransferBridgeInterface $releaseGroupManager,
-        PackageManagementSystemBridge $dataProviderManager,
+        ComposerUpdateStrategy $composerUpdateStrategy,
```

Comment:
our development should be based on the Interfaces, not on the realizations
in the *__construct*
we should use
\Upgrader\Business\Upgrader\Strategy\UpgradeStrategyInterface

Reply:
Thanks, replace by Interfaces